lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -209,15 +209,7 @@ def add_included_resources_for(serializer, include_tree, primary_data, included)
 
         def links_for(serializer)
           serializer.links.each_with_object({}) do |(name, value), hash|
-            hash[name] =
-              if value.respond_to?(:call)
-                link = Link.new(serializer)
-                link.instance_eval(&value)
-
-                link.to_hash
-              else
-                value
-              end
+            hash[name] = Link.new(serializer).value(value)
```

Comment:
:+1: to moving this logic inside the `Link` class, which is in line with what we've done with `Attribute`/`Reflection`.